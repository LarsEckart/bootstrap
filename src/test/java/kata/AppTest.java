package kata;

import org.approvaltests.Approvals;
import org.approvaltests.StoryBoard;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.intellij.IntelliJReporter;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
@UseReporter(IntelliJReporter.class)
class AppTest {

    @Test
    void testBasicPlay() {
        StoryBoard story = new StoryBoard();
        Skyjo skyjo = new Skyjo(new Player("Alice"), new Player("Bob"));
        story.addFrame(skyjo.toString());
        skyjo.startGame();
        story.addFrame(skyjo.toString());
        skyjo.flipInitialCards();
        story.addFrame(skyjo.toString());

        Approvals.verify(story);
    }
}
