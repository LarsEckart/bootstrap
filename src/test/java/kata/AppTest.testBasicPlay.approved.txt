Initial:
Deck: 150
Players:
Alice
no cards yet

Bob
no cards yet



Frame #1:
Deck: 126
Players:
Alice
<X1X><X1X><X11X><X12X>
<X8X><X-2X><X3X><X4X>
<X10X><X12X><X0X><X9X>

Bob
<X2X><X3X><X0X><X2X>
<X4X><X9X><X-2X><X2X>
<X11X><X6X><X0X><X10X>



Frame #2:
Deck: 126
Players:
Alice
<1><1><X11X><X12X>
<X8X><X-2X><X3X><X4X>
<X10X><X12X><X0X><X9X>

Bob
<2><3><X0X><X2X>
<X4X><X9X><X-2X><X2X>
<X11X><X6X><X0X><X10X>

